src/http_server.h

```diff
@@ -362,6 +362,15 @@ class HTTPAPIServer : public HTTPServer {
       const int32_t port, const bool reuse_port, const std::string& address,
       const std::string& header_forward_pattern, const int thread_cnt,
       const RestrictedFeatures& restricted_apis);
```

Comment:
one other way to remove the need for a separate constructor would be to remove the const& and add a default value for restricted_apis.